ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ExecuteQueryResponseHandler.java

```diff
@@ -61,14 +67,31 @@ protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata)
 
   @Override
   protected void handleRow(final Buffer buff) {
-    final JsonArray values = new JsonArray(buff);
-    if (rows.size() < maxRows) {
-      rows.add(new RowImpl(columnNames, columnTypes, values, columnNameToIndex));
+    final Row row;
+    final Object json = buff.toJson();
+
+    if (json instanceof JsonObject) {
+      final JsonObject jsonObject = (JsonObject) json;
+      // This is the serialized consistency vector
+      // Don't add it to the result list since the user should not see it
```

Comment:
We talked before about supporting an older client.  It seems like the server will just return the consistency token if it's enabled on the server.  We should only send it down if it's enabled on the client which can be done by indicating that it's enabled in the request property.